src/main/java/dev/j3fftw/litexpansion/Events.java

```diff
@@ -68,6 +79,37 @@ public void onHungerDamage(EntityDamageEvent e) {
         }
     }
 
+    @EventHandler
+    @SuppressWarnings("ConstantConditions")
+    public void onGlassCut(PlayerInteractEvent e) {
+        final Block block = e.getClickedBlock();
+        if (block == null) return;
+
+        final Material blockType = block.getType();
+        final Location blockLocation = block.getLocation();
+        final GlassCutter glassCutter = (GlassCutter) SlimefunItem.getByID(Items.GLASS_CUTTER.getItemId());
+
+        if ((blockType == Material.GLASS
+            || blockType == Material.GLASS_PANE
+            || blockType.name().endsWith("_GLASS")
+            || blockType.name().endsWith("_GLASS_PANE")
+        ) && glassCutter.isItem(e.getItem())
```

Comment:
```suggestion
        ) && item != null && item.getID().equals(Items.GLASS_CUTTER.getItemId()))
```

Reply:
heard isItem is better, not 100% sure in which cases

Reply:
Hmm. Maybe it would be a better approach to create a GlassCutter field and initialize in the constructor, then you could just do ``glassCutter.isItem(e.getItem())``

Reply:
isnt that whats already currently done?

Reply:
Nope. "create a GlassCutter field and initialize in the constructor"

Reply:
Ima need help on how to do that, could u explain pls o.o

Reply:
```java
public class Events implements Listener {

    private final GlassCutter glassCutter;

    public Events() {
        this.glassCutter = (GlassCutter) Items.GLASS_CUTTER.getItem();
    }

```

Reply:
Isnt this longer than whats already done? I'm not sure I understand why this is better

Reply:
Also that has the same effect as just moving `final GlassCutter glassCutter = (GlassCutter) Items.GLASS_CUTTER.getItem();` to the top, doesnt it?

Reply:
Because you getting that item on every interact event...

Reply:
> Also that has the same effect as just moving final GlassCutter glassCutter = (GlassCutter) Items.GLASS_CUTTER.getItem(); to the top, doesnt it?

I think yes.

Reply:
Alright, did that, gonna mark as resolved